app/controllers/shelters_controller.rb

```diff
@@ -32,14 +31,29 @@ def update
   end
 
   def destroy
-    Shelter.destroy(params[:id])
+    shelter = Shelter.find(params[:id])
+    remove_favorited_pets(shelter)
+    shelter.destroy
     redirect_to '/shelters'
   end
 
   private
 
   def shelter_params
-    params.permit(:name, :address, :city, :state, :zip)
+    params.permit(:name,
+                  :address,
+                  :city,
+                  :state,
+                  :zip)
+  end
+
+  def remove_favorited_pets(shelter)
+    pets = shelter.pets.map { |pet| "#{pet.id}" }
```

Comment:
I'm a little confused why we need to interpolate in the block? Can't this line just be written as `pets = shelter.pets.map(&:id)` ?

Reply:
Yep. It was the first thing that came to me and it worked so I left it as is lol